### ISSUE TITLE
Test failure on Jenkins - Capybara::Poltergeist::JavascriptError

### ISSUE THREAD

**Issue by chrisroos** [alphagov/smart-answers#1624]
I saw this error in [Jenkins build 2378][jenkins-2378]. It appears to be intermittent as the following build succeeded.

The tests were run with seed 53125.

[jenkins-2378]: https://ci-new.alphagov.co.uk/job/govuk_smart_answers/2378/consoleFull

```
ReportAProblemTest
  test_: with javascript smart-answer start page should have current url in report-a-problem. ERROR (0.11s)
Capybara::Poltergeist::JavascriptError:         Capybara::Poltergeist::JavascriptError: One or more errors were raised in the Javascript code on the page. If you don't care about these errors, you can ignore them by setting js_errors: false in your Poltergeist configuration (see documentation for details).
        
        TypeError: 'null' is not an object (evaluating 'data['title']')
        TypeError: 'null' is not an object (evaluating 'data['title']')
            at http://127.0.0.1:50084/smartanswers/smart-answers.js:97 in addToHistory
            at http://127.0.0.1:50084/smartanswers/smart-answers.js:89
            at https://assets.digital.cabinet-office.gov.uk/static/libs/jquery/jquery-1.7.2.js:16
            at https://assets.digital.cabinet-office.gov.uk/static/libs/jquery/jquery-1.7.2.js:16
            at https://assets.digital.cabinet-office.gov.uk/static/libs/jquery/jquery-1.7.2.js:18 in S
            at https://assets.digital.cabinet-office.gov.uk/static/libs/jquery/jquery-1.7.2.js:18
            /home/jenkins/bundles/govuk_smart_answers/ruby/2.1.0/gems/poltergeist-1.6.0/lib/capybara/poltergeist/browser.rb:323:in `command'
            /home/jenkins/bundles/govuk_smart_answers/ruby/2.1.0/gems/poltergeist-1.6.0/lib/capybara/poltergeist/browser.rb:31:in `visit'
            /home/jenkins/bundles/govuk_smart_answers/ruby/2.1.0/gems/poltergeist-1.6.0/lib/capybara/poltergeist/driver.rb:95:in `visit'
            /home/jenkins/bundles/govuk_smart_answers/ruby/2.1.0/gems/capybara-2.1.0/lib/capybara/session.rb:193:in `visit'
            /home/jenkins/bundles/govuk_smart_answers/ruby/2.1.0/gems/capybara-2.1.0/lib/capybara/dsl.rb:51:in `block (2 levels) in <module:DSL>'
            /home/jenkins/workspace/govuk_smart_answers/test/integration/engine/report_a_problem_test.rb:8:in `block (3 levels) in <class:ReportAProblemTest>'
            /home/jenkins/bundles/govuk_smart_answers/ruby/2.1.0/gems/shoulda-context-1.2.1/lib/shoulda/context/context.rb:436:in `instance_exec'
            /home/jenkins/bundles/govuk_smart_answers/ruby/2.1.0/gems/shoulda-context-1.2.1/lib/shoulda/context/context.rb:436:in `block in run_current_setup_blocks'
            /home/jenkins/bundles/govuk_smart_answers/ruby/2.1.0/gems/shoulda-context-1.2.1/lib/shoulda/context/context.rb:434:in `each'
            /home/jenkins/bundles/govuk_smart_answers/ruby/2.1.0/gems/shoulda-context-1.2.1/lib/shoulda/context/context.rb:434:in `run_current_setup_blocks'
            /home/jenkins/bundles/govuk_smart_answers/ruby/2.1.0/gems/shoulda-context-1.2.1/lib/shoulda/context/context.rb:426:in `run_all_setup_blocks'
            /home/jenkins/bundles/govuk_smart_answers/ruby/2.1.0/gems/shoulda-context-1.2.1/lib/shoulda/context/context.rb:430:in `run_parent_setup_blocks'
            /home/jenkins/bundles/govuk_smart_answers/ruby/2.1.0/gems/shoulda-context-1.2.1/lib/shoulda/context/context.rb:402:in `block in create_test_from_should_hash'
        /home/jenkins/bundles/govuk_smart_answers/ruby/2.1.0/gems/poltergeist-1.6.0/lib/capybara/poltergeist/browser.rb:323:in `command'
        /home/jenkins/bundles/govuk_smart_answers/ruby/2.1.0/gems/poltergeist-1.6.0/lib/capybara/poltergeist/browser.rb:31:in `visit'
        /home/jenkins/bundles/govuk_smart_answers/ruby/2.1.0/gems/poltergeist-1.6.0/lib/capybara/poltergeist/driver.rb:95:in `visit'
        /home/jenkins/bundles/govuk_smart_answers/ruby/2.1.0/gems/capybara-2.1.0/lib/capybara/session.rb:193:in `visit'
        /home/jenkins/bundles/govuk_smart_answers/ruby/2.1.0/gems/capybara-2.1.0/lib/capybara/dsl.rb:51:in `block (2 levels) in <module:DSL>'
        /home/jenkins/workspace/govuk_smart_answers/test/integration/engine/report_a_problem_test.rb:8:in `block (3 levels) in <class:ReportAProblemTest>'
        /home/jenkins/bundles/govuk_smart_answers/ruby/2.1.0/gems/shoulda-context-1.2.1/lib/shoulda/context/context.rb:436:in `instance_exec'
        /home/jenkins/bundles/govuk_smart_answers/ruby/2.1.0/gems/shoulda-context-1.2.1/lib/shoulda/context/context.rb:436:in `block in run_current_setup_blocks'
        /home/jenkins/bundles/govuk_smart_answers/ruby/2.1.0/gems/shoulda-context-1.2.1/lib/shoulda/context/context.rb:434:in `each'
        /home/jenkins/bundles/govuk_smart_answers/ruby/2.1.0/gems/shoulda-context-1.2.1/lib/shoulda/context/context.rb:434:in `run_current_setup_blocks'
        /home/jenkins/bundles/govuk_smart_answers/ruby/2.1.0/gems/shoulda-context-1.2.1/lib/shoulda/context/context.rb:426:in `run_all_setup_blocks'
        /home/jenkins/bundles/govuk_smart_answers/ruby/2.1.0/gems/shoulda-context-1.2.1/lib/shoulda/context/context.rb:430:in `run_parent_setup_blocks'
        /home/jenkins/bundles/govuk_smart_answers/ruby/2.1.0/gems/shoulda-context-1.2.1/lib/shoulda/context/context.rb:402:in `block in create_test_from_should_hash'
```

**Comment by chrisroos** [comment]
I've just successfully run the tests locally using the same seed (`TESTOPTS="--seed=53125" bundle exec rake`) so I don't think it's a seed related problem.

**Comment by chrisroos** [comment]
This test failed again in [Jenkins build 2420](https://ci-new.alphagov.co.uk/job/govuk_smart_answers/2420/console), using a seed of 41067.

**Comment by chrisroos** [comment]
This test failed again in [Jenkins build 2430](https://ci-new.alphagov.co.uk/job/govuk_smart_answers/2430/console), using a seed of 58458.

**Comment by floehopper** [comment]
This test failed again in [Jenkins build 2439](https://ci-new.alphagov.co.uk/job/govuk_smart_answers/2439/console), using a seed of 45995.

**Comment by floehopper** [comment]
This test failed again in [Jenkins build 2543](https://ci-new.alphagov.co.uk/job/govuk_smart_answers/2543/), using a seed of 19961.

**Comment by floehopper** [comment]
This test failed again in [Jenkins build 2547](https://ci-new.alphagov.co.uk/job/govuk_smart_answers/2547/), using a seed of 5818.

**Comment by floehopper** [comment]
This test failed again in [Jenkins build 2367](https://ci-new.alphagov.co.uk/job/govuk_smartanswers_branches/2367/), using a seed of 63776.

**Comment by floehopper** [comment]
I've just merged my attempted fix for this in 1d78712ad821f58909c38f3e1d2e8a7ed9205f45. If we see the problem again, we should probably revert that commit.

**Comment by floehopper** [comment]
Unfortunately I've just seen this failure occur on my local machine even though the attempted fix was in place, so we should probably revert that attempted fix. :disappointed:

**Comment by floehopper** [comment]
I've reverted the attempted fix in dee8c2b5f925c8f36f3b96bcf300f9565cdc7833.

**Comment by chrisroos** [comment]
I've moved this to Trello in an attempt to end up with a single place containing Smart Answers stories/tasks/bugs.

See the [Trello card](https://trello.com/c/nG7qgyLz/138-test-failure-on-jenkins-capybara-poltergeist-javascripterror) (not public).

Closing this issue.
